signer/src/transaction_coordinator.rs

```diff
@@ -288,20 +291,61 @@ where
         let account = stacks.get_account(wallet.address()).await?;
         wallet.set_nonce(account.nonce);
 
-        // Generate
-        let _sign_requests = futures::stream::iter(deposit_requests)
-            .then(|req| self.construct_stacks_sign_request(req, bitcoin_aggregate_key, &wallet))
-            .try_collect::<Vec<StacksTransactionSignRequest>>()
-            .await?;
+        for req in deposit_requests {
+            let sign_request_fut =
+                self.construct_stacks_sign_request(req, bitcoin_aggregate_key, &wallet);
+
+            let (sign_request, multi_tx) = match sign_request_fut.await {
+                Ok(res) => res,
+                Err(error) => {
+                    tracing::error!(%error, "could not construct a transaction completing the deposit request");
+                    continue;
+                }
+            };
+
+            // If we fail here, then we need to decrement the nonce.
+            let process_request_fut =
+                self.process_sign_request(sign_request, chain_tip, multi_tx, &wallet);
+
+            match process_request_fut.await {
+                Ok(txid) => {
+                    tracing::info!(%txid, "successfully submitted complete-deposit transaction")
+                }
+                Err(error) => {
+                    tracing::warn!(%error, "could process the sign request for deposit request");
+                    wallet.set_nonce(wallet.get_nonce().saturating_sub(1));
+                }
+            }
+        }
 
-        // TODO:
-        // 1. Broadcast the sign requests
-        // 2. Gather the signatures into the transaction.
-        // 3. Broadcast the transaction to the stacks network. Then go home
-        //    and relax.
         Ok(())
     }
 
+    /// Sign and broadcast the stacks transaction
+    ///
+    /// If we fail to sign the transaction for some reason, we decrement
+    /// the nonce by one, and try the next transaction. This is not a fatal
+    /// error, since we could fail to sign the transaction because someone
+    /// else is now the coordinator, and all of the signers are now
+    /// ignoring us.
```

Comment:
nit: move this part above? It seems out of place here, since we are not decrementing the nonce (here)

Reply:
Done in https://github.com/stacks-network/sbtc/pull/617/commits/7cdea8d577b0e2a7d7ed9335e26d38e87252a0e9